lib/CPathfinder.cpp

```diff
@@ -479,25 +558,24 @@ CPathsInfo::~CPathsInfo()
 	delete [] nodes;
 }
 
-const CGPathNode * CPathsInfo::getPathInfo( int3 tile ) const
+const CGPathNode * CPathsInfo::getPathInfo(const int3 &tile, const EPathfindingLayer &layer) const
 {
 	boost::unique_lock<boost::mutex> pathLock(pathMx);
 
-	if(tile.x >= sizes.x || tile.y >= sizes.y || tile.z >= sizes.z)
+	if(tile.x >= sizes.x || tile.y >= sizes.y || tile.z >= sizes.z || layer >= EPathfindingLayer::NUM_LAYERS)
```

Comment:
In which cases this check is necessary? Silent failure may result in hard-to-track errors later. I suggest to replace these with assert's and fix caller's code if necessary. And BTW - calling this code with int3(-1, -1, -1) will cause crash since int3 is signed.

Yes, this code was here before but this doesn't mean that it shouldn't be fixed. Especially during pathfinder redo.